bfextractor.py

```diff
@@ -42,13 +42,47 @@ def build_pyramid(img, n):
 def transform_xml(metadata, id_map):
     ElementTree.register_namespace('', OME_NS)
     xml_root = ElementTree.fromstring(metadata.dumpXML())
+    missing = []
     iterfind = lambda m: xml_root.iterfind(f'ome:{m}', {'ome': OME_NS})
+    # Replace original image ID attributes with our UUID-based IDs.
     for elt in itertools.chain(iterfind('Image'), iterfind('ImageRef')):
-        elt.attrib['ID'] = image_id_map[elt.attrib['ID']]
+        new_id = id_map.get(elt.attrib['ID'])
+        if new_id is not None:
+            elt.attrib['ID'] = new_id
+        else:
+            # Keep a list of elements missing from the map.
+            missing.append(elt)
+    # Drop the missing elements. This is currently only meant to handle Faas
+    # pyramids where we need to drop the subresolution images.
+    for elt in missing:
+        # We should only ever see Image elements in this list.
+        if elt.tag == f'{{{OME_NS}}}Image':
+            xml_root.remove(elt)
+        else:
+            raise ValueError('Unexpected reference to dropped image')
     xml_str = ElementTree.tostring(xml_root, encoding='utf-8')
     return xml_str
 
 
+def is_pyramid(image_reader):
```

Comment:
Conceivably we might want to detect pyramids for other purposes (such as optimising tile creation) so maybe call this `is_faas_pyramid`?

Reply:
Agreed.